### PR TITLE
feat(backend, frontend): publication can be disabled/workspace

### PR DESCRIPTION
**Pull request by grignards**
See #4270.

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend_app_workspace_advanced/src/container/WorkspaceAdvanced.jsx [hunk 1]
```diff
@@ -417,6 +418,29 @@ export class WorkspaceAdvanced extends React.Component {
...
+  handleTogglePublicationEnabled = async () => {
+    const { props, state } = this
+    const newPublicationEnabledValue = !state.content.publication_enabled
+
+    const fetchTogglePublicationEnabled = await handleFetchResult(await putPublicationEnabled(state.config.apiUrl, state.content, newPublicationEnabledValue))
+
+    switch (fetchTogglePublicationEnabled.apiResponse.status) {
+      case 200:
+        this.sendGlobalFlashMessage(
+          newPublicationEnabledValue ? props.t('Publication activated') : props.t('Publication deactivated'),
```
Comment (gbonaspetti) [inline frontend_app_workspace_advanced/src/container/WorkspaceAdvanced.jsx:430]: Do you add the translations?

Comment (grignards) [reply frontend_app_workspace_advanced/src/container/WorkspaceAdvanced.jsx]: Done.

File: frontend_app_workspace_advanced/i18next.scanner/fr/translation.json [hunk 2]
```diff
@@ -50,5 +50,9 @@
   "Request accepted": "Demande acceptée",
   "Request rejected": "Demande refusée",
   "There are no requests yet.": "Vous n'avez pas encore de demande.",
-  "This is the role that members will have by default when they join your space (for open and on request spaces only).": "C'est le rôle que les membres auront par défaut lorsqu'ils rejoignent votre espace (uniquement pour les espaces ouverts ou sur demande)."
+  "This is the role that members will have by default when they join your space (for open and on request spaces only).": "C'est le rôle que les membres auront par défaut lorsqu'ils rejoignent votre espace (uniquement pour les espaces ouverts ou sur demande).",
+  "Publications activated": "Publications activées",
+  "Publications deactivated": "Publications desactivées",
+  "Error while activating publications": "Erreur pendant l'activation des publications",
+  "Error while deactivating publications": "Erreur pendant l'activation des publications"
```
Comment (gbonaspetti) [inline frontend_app_workspace_advanced/i18next.scanner/fr/translation.json:57]:
```suggestion
  "Error while deactivating publications": "Erreur pendant la désactivation des publications"
```

File: frontend_app_workspace_advanced/i18next.scanner/pt/translation.json [hunk 3]
```diff
@@ -50,5 +50,9 @@
   "Request accepted": "Pedido aceito",
   "Request rejected": "Pedido rejeitado",
   "There are no requests yet.": "Ainda não tem pedidos.",
-  "This is the role that members will have by default when they join your space (for open and on request spaces only).": "Esta é a função que os membros terão por pré-definição quando se juntarem ao seu espaço (apenas para espaços abertos ou a pedido)."
+  "This is the role that members will have by default when they join your space (for open and on request spaces only).": "Esta é a função que os membros terão por pré-definição quando se juntarem ao seu espaço (apenas para espaços abertos ou a pedido).",
+  "Publications activated": "__NOT_TRANSLATED__",
+  "Publications deactivated": "__NOT_TRANSLATED__",
+  "Error while activating publications": "__NOT_TRANSLATED__",
+  "Error while deactivating publications": "__NOT_TRANSLATED__"
```
Comment (gbonaspetti) [inline frontend_app_workspace_advanced/i18next.scanner/pt/translation.json:57]:
```suggestion
  "Publications activated": "Publicações ativadas",
  "Publications deactivated": "Publicações desativadas",
  "Error while activating publications": "Erro ao activar publicações",
  "Error while deactivating publications": "Erro ao desactivar publicações"
```

**Review by gbonaspetti — APPROVED** [review]

File: frontend/src/container/Dashboard.jsx [hunk 4]
```diff
@@ -437,15 +437,20 @@ export class Dashboard extends React.Component {
 
     const userRoleIdInWorkspace = findUserRoleIdInWorkspace(props.user.userId, props.curWs.memberList, ROLE_LIST)
 
-    // INFO - GB - 2019-08-29 - these filters are made temporarily by the frontend, but may change to have all the intelligence in the backend
+    // INFO - GB - 2019-08-29 - these filters are made temporarly by the frontend, but may change to have all the intelligence in the backend
```
Comment (raphj) [inline frontend/src/container/Dashboard.jsx:440]:
```suggestion
    // INFO - GB - 2019-08-29 - these filters are made temporarily by the frontend, but may change to have all the intelligence in the backend
```
There is a "i" in temporarily :-)

File: frontend_app_workspace_advanced/i18next.scanner/fr/translation.json (hunk 2 above)
Comment (raphj) [inline frontend_app_workspace_advanced/i18next.scanner/fr/translation.json:57]:
```suggestion
  "Error while deactivating publications": "Erreur pendant la désactivation des publications"
```

Comment (grignards) [reply frontend_app_workspace_advanced/i18next.scanner/fr/translation.json]: Already fixed by giulia's suggestion.

File: frontend_lib/src/helper.js [hunk 5]
```diff
@@ -781,3 +781,9 @@ export const getAvatarBaseUrl = (apiUrl, userId) => `${apiUrl}/users/${userId}/a
 export const getCoverBaseUrl = (apiUrl, userId) => `${apiUrl}/users/${userId}/cover`
 
 export const getFileDownloadUrl = (apiUrl, workspaceId, contentId, filename) => `${apiUrl}/workspaces/${workspaceId}/files/${contentId}/raw/${filename}?force_download=1`
+
+/**
+ * DOC - SG - 2021-04-15
+ * Return the string with its first letter in uppercase.
+ */
+export const capitalize = (string) => string[0].toUpperCase() + string.slice(1)
```
Comment (raphj) [inline frontend_lib/src/helper.js:789]: This probably breaks emojis

**Review by gbonaspetti — APPROVED** [review]
frontend - approved

File: frontend_lib/src/helper.js (hunk 5 above)
Comment (raphj) [reply frontend_lib/src/helper.js:789]: Where is it used? I can't manage to find it.

File: frontend/src/container/Dashboard.jsx (hunk 4 above)
Comment (grignards) [reply frontend/src/container/Dashboard.jsx]: Corrected.

File: frontend_lib/src/helper.js (hunk 5 above)
Comment (grignards) [reply frontend_lib/src/helper.js]: In the end it is not used, I removed it.

**Review by raphj — APPROVED** [review]

**Pull request merged**